packages/create-cloudflare/src/templates.ts

```diff
@@ -80,12 +88,24 @@ export type TemplateConfig = {
 	path?: string;
 };
 
+type CopyFiles = (StaticFileMap | VariantInfo) & {
+	destinationDir?: string | ((ctx: C3Context) => string);
+};
+
 // A template can have a number of variants, usually js/ts
 type VariantInfo = {
 	path: string;
 };
 
-type StaticFileMap = Record<string, VariantInfo>;
+type StaticFileMap = {
+	selectVariant?: (ctx: C3Context) => Promise<string>;
+	variants: Record<string, VariantInfo>;
+};
+
+const defaultSelectVariant = async (ctx: C3Context) => {
+	const typescript = await shouldUseTs(ctx);
```

Comment:
There's a pretty subtle existing bug here that I didn't notice until I was reviewing your PR. It's not introduced by your change, but I'm wondering if you can fix it since you're touching this code path. 

If you run this code with a pages template that has a `copyFiles` definition and select `js` during the process, you'll be prompted if you want to use typescript by c3. 

You can repro this by running the solid template and selecting not to use `ts`. Here's what happens:
![image](https://github.com/cloudflare/workers-sdk/assets/204386/8a6a9151-6ee9-4324-aaff-a51b741d0537).

I think to fix this, we need to adjust `shouldUseTs` so that it returns the result of `usesTypescript` if `ctx.platform === 'pages'`.

Reply:
done 🙂